src/pymodaq_plugins_keithley/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Keithley27XX.py

```diff
@@ -0,0 +1,248 @@
+import os
+import numpy as np
+from easydict import EasyDict as edict
+from pymodaq.utils.daq_utils import ThreadCommand
+from pymodaq.utils.data import DataFromPlugins, DataToExport
+from pymodaq.control_modules.viewer_utility_classes import DAQ_Viewer_base, comon_parameters, main
+from pymodaq.utils.parameter import Parameter
+
+import pymodaq_plugins_keithley as plugin
+from pymodaq_plugins_keithley.hardware.keithley27XX.keithley27XX_VISADriver import Keithley27XXVISADriver as Keithley
+
+
+class DAQ_0DViewer_Keithley27XX(DAQ_Viewer_base):
+    """ Keithley plugin class for a OD viewer.
+
+    This object inherits all functionalities to communicate with PyMoDAQ’s DAQ_Viewer module through inheritance via
+    DAQ_Viewer_base. It makes a bridge between the DAQ_Viewer module and the Python wrapper of a the keithley 2700.
+
+    * Keithley 27XX should be compatible with this plugin
+    * Tested with Keithley Multimeter/Switch System 2700 - 2701
+    * PyMoDAQ version = 4.0.11 during the test
+
+    :param controller: The particular object that allow the communication with the hardware, in general a python wrapper around the hardware library.
+    :type  controller:  object
+
+    :param params: Parameters displayed in the daq_viewer interface
+    :type params: dictionnary list
+
+    :param x_axis: Daq_0Dviewer doesn't have axis, data acquired sample by sample
+    :type x_axis: Nonetype
+    """
+    all_config = {}
+    current_config = None
+    resources_path = plugin.__path__[0] + "/resources"
+
+    # Configurations for supported Keithley instruments
+    toml_keithley = [f for f in os.listdir(resources_path) if "keithley.toml" in f]
+    all_config["base"] = plugin.config_keithley
+
+    # Configurations for supported Keithley switching modules
+    toml_modules = [f for f in os.listdir(resources_path) if "module" in f and ".toml" in f]
+    for file in toml_modules:
+        exec("all_config[" + str(file[-9:-5]) + "] = plugin.config_k" + str(file[-9:-5]))
+        if current_config == None:
+            current_config = all_config.get(int(file[-9:-5]))
+
+    # Read configuration file
+    panel = all_config["base"]('INSTRUMENT').get('panel').upper()
+    resources_list = [rsrc("INSTRUMENT").get("rsrc_name") for key, rsrc in list(all_config.items()) if key != "base"]
+    print("resources list = ", resources_list)
+
+    if panel == 'FRONT':
```

Comment:
why not grouping your Front and Rear settings in a dedicated group and showing/hiding it given the config or choice of the user?

Reply:
Looks a good idea, I will try to do it soon